### PR TITLE
Fixed #4566: Constructor added for LayerGradient

### PR DESCRIPTION
**Pull request by pandamicro**

### CODE REVIEW

File: cocos/2d/CCLayer.cpp [hunk 1]
```diff
@@ -629,7 +629,20 @@ void LayerColor::onDraw(const kmMat4& transform, bool transformUpdated)
 }
 //
 // LayerGradient
-// 
+//
+LayerGradient::LayerGradient()
+: _startColor(Color4B(0, 0, 0, 255))
+, _endColor(Color4B(0, 0, 0, 255))
```
Comment (dumganhar) [inline cocos/2d/CCLayer.cpp]: Please use `Color4B::BLACK` instead of hard code.

File: cocos/2d/CCLayer.cpp [hunk 2]
```diff
@@ -629,7 +629,20 @@ void LayerColor::onDraw(const kmMat4& transform, bool transformUpdated)
...
+, _startOpacity(255)
+, _endOpacity(255)
+, _alongVector(Point(0, -1))
```
Comment (dumganhar) [inline cocos/2d/CCLayer.cpp:639]: Why to init `_alongVector` to `Point(0, -1)`?

File: cocos/2d/CCLayer.cpp [hunk 3]
```diff
@@ -629,7 +629,20 @@ void LayerColor::onDraw(const kmMat4& transform, bool transformUpdated)
...
+, _compressedInterpolation(true)
+{
+    
+}
+LayerGradient::~LayerGradient() {
```
Comment (dumganhar) [inline cocos/2d/CCLayer.cpp]: `{` needs to be at the next line.

File: cocos/2d/CCLayer.cpp (hunk 2 above)
Comment (pandamicro) [inline cocos/2d/CCLayer.cpp:639]: Because it was inited like so when vector is not provided in initWithColor function

**Comment by pandamicro** [comment]
Thanks

**Pull request merged**